### ISSUE TITLE
AccordionPanel: Slide Behavior IE11 Flickering

### ISSUE THREAD

**Issue by hink084** [primefaces/primefaces#4631]
The execution order of the slide up / down behavior within the Accordion Panel has changed.  You can observe the execution order by setting a breakpoint within the Accordion Panel JavaScript (see below).  This change of behavior is causing flickering in IE, most likely due to the content of the page briefly becoming larger than expected and then potentially showing / hiding an unnecessary vertical scrollbar on the page body.  I believe the cause of this is related to the jQuery v3 upgrade.

## 1) Environment
- PrimeFaces version: 7.0-RC3 - Latest Master
- Does it work on the newest sources in GitHub? No
- Affected browsers: IE11

## 2) Expected behavior
Click on an Accordion Panel tab --> the previously open panel slides up --> the new panel slides down.

## 3) Actual behavior
Click on an Accordion Panel tab --> the new panel slides down --> the previously open panel slides up.

## 4) Steps to reproduce
You can observe this behavior using any browser in the PrimeFaces Showcase AccordionPanel:
https://www.primefaces.org/showcase/ui/panel/accordionPanel.xhtml

In the accordion.js (or components.js file within the browser) set breakpoints on the following lines:
```
.next().attr('aria-hidden', true).slideUp(function(){
```
```
panel.attr('aria-hidden', false).slideDown('normal', function() {
``` 

##### accordion.js:
```javascript
    show: function(panel) {
        var _self = this;

        //deactivate current
        if(!this.cfg.multiple) {
            var oldHeader = this.headers.filter('.ui-state-active');
            oldHeader.children('.ui-icon').removeClass(this.cfg.expandedIcon).addClass(this.cfg.collapsedIcon);
            oldHeader.attr('aria-selected', false);
            oldHeader.attr('aria-expanded', false).removeClass('ui-state-active ui-corner-top').addClass('ui-corner-all')
                .next().attr('aria-hidden', true).slideUp(function(){
                    if(_self.cfg.onTabClose)
                        _self.cfg.onTabClose.call(_self, panel);
                });
        }

        //activate selected
        var newHeader = panel.prev();
        newHeader.attr('aria-selected', true);
        newHeader.attr('aria-expanded', true).addClass('ui-state-active ui-corner-top').removeClass('ui-state-hover ui-corner-all')
                .children('.ui-icon').removeClass(this.cfg.collapsedIcon).addClass(this.cfg.expandedIcon);

        panel.attr('aria-hidden', false).slideDown('normal', function() {
            _self.postTabShow(panel);
        });
    },
```

After stepping over the two breakpoints, you will see the effect of the slide up is not executed immediately; however, the effect of the slide down is instantaneous - hence causing the slide down effect to occur prior to the slide up effect.  In PrimeFaces v6.1, neither side up nor slide down were executed immediately.  Instead they were placed in some sort of queue in which the slide up effect occurred prior to the slide down effect as expected.

As of right now, I'm unsure what is causing this...

**Comment by melloware** [comment]
I wonder if the difference is slideDown is passed "normal" but slideUp is not passed that effect param?

However according to the docs "normal" and omitted should default to 400ms: http://api.jquery.com/slideup/
http://api.jquery.com/slidedown/

And it looks like queued=true is the default.

**Comment by hink084** [comment]
So I'm still navigating down the debugging rabbit hole, but I think this is a jQuery timing issue rather than an animation issue - in other words, I don't believe the duration of the slideDown and slideUp is at fault.  

However, the "normal" argument passed in the PrimeFaces accordion.js is erroneous, but as you mentioned it shouldn't effect anything and default to the 400ms ('fast' or 'slow' are the only valid String arguments).

I tested the Accordion on IE11 with:
* PrimeFaces 6.1 with jQuery 1.11.3
* PrimeFaces 6.2 with jQuery 3.2.1
* PrimeFaces 7.0RC3 with jQuery 3.3.1

and the flickering started with PrimeFaces 6.2 with the jQuery 3.2.1 upgrade as suspected.

One resolution is for the developer to just disable all animations in IE via:
https://api.jquery.com/jquery.fx.off/
```
jQuery.fx.off = true
```

At the moment, debugging why the slideDown is occurring before the slideUp has become a little confusing to me.  If you skip over the slideUp breakpoint and then continue onward, the slideDown effect will occur before the slideUp effect.  However, if I step into the slideUp breakpoint and continue stepping through it, the slideUp will actually occur before the slideDown.  

Ultimately, I think this is going to be a jQuery problem and not a PrimeFaces problem, but the Accordion does have a flickering issue with IE11 because of it.

**Comment by melloware** [comment]
Great debugging.  And this only affects IE11?  Definitely feels like something Jquery changed between 1.11 and the 3.X branch.

**Comment by hink084** [comment]
I've been debugging with Chrome, but I've only seen flickering issues with IE11. I've also tested Firefox and Samsung Internet Browser and didn't see any flickering.

**Comment by melloware** [comment]
Looks like this Jquery bug is the culprit: https://github.com/jquery/jquery/issues/4113

Marking this as "thirdparty" and hopefully it will be fixed in Jquery 4.0.
